File: src/java/magma/app/Location.java
package magma.app;

import magma.api.collect.Iter;
import magma.api.collect.list.List;
import magma.api.text.Strings;

public record Location(List<String> namespace, String name) {
    public String attachExtension(String extension) {
        return this.name + "." + extension;
    }

    Iter<String> iterNamespace() {
        return this.namespace.iter();
    }

    public boolean hasName(String name) {
        return Strings.equalsTo(this.name, name);
    }
}

File: src/ts/magma/app/Location.ts
import { List } from "../../magma/api/collect/list/List";
import { Iter } from "../../magma/api/collect/Iter";
import { Strings } from "../../magma/api/text/Strings";
export class Location {
	namespace: List<string>;
	name: string;
	constructor (namespace: List<string>, name: string) {
		this.namespace = namespace;
		this.name = name;
	}
	attachExtension(extension: string): string {
		return this.name + "." + extension/*unknown*/;
	}
	iterNamespace(): Iter<string> {
		return this.namespace.iter()/*unknown*/;
	}
	hasName(name: string): boolean {
		return Strings.equalsTo(this.name, name)/*unknown*/;
	}
}
